apple/internal/apple_framework_import.bzl

```diff
@@ -249,7 +249,7 @@ def _apple_dynamic_framework_import_impl(ctx):
         "dynamic_framework_file",
         header_imports,
         module_map_imports,
-        _all_framework_binaries(framework_groups),
+        [] if ctx.attr.avoid_link else _all_framework_binaries(framework_groups),
```

Comment:
Does this work with frameworks built from source and passed with `ios_framework` ? I think this would be a welcoming change 

My current ad-hoc implementation of framework integration has a few pieces:
- a "framework middle man" rule
- a "framework bundler" rule + aspect
The framework middle man is a way to attach frameworks to a rule and the framework bundler handles the business logic of how these frameworks are treated by the build: theres a few possibilites: embedding, embedding+linking, embedding+side-loaded-rpath. It'd be cool to upstream all of this and would probably be better if it wasn't add hoc 👍

Reply:
this is only handling it with apple_dynamic_framework_import